File: 53maxSubArray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        # Initialize max_sum with the first value from nums
        max_sum = nums[0]
        cur_sum = 0
        for n in nums:
            # If sum is ever negative disregard and start over since 0 is always greater than a negative sum
            if cur_sum < 0:
                cur_sum = 0
            # Update the current sum
            cur_sum += n
            # Update max_sum
            max_sum = max(max_sum, cur_sum)
        return max_sum

File: Java/53maxSubArray.java
class Solution {
    public int maxSubArray(int[] nums) {
        // Initialize maxSum to the first position in the array
        int maxSum = nums[0];
        // Set the currSum to 0
        int currSum = 0;
        for (int n : nums) {
            // If currSum is less than 0 disregard and start over
            if (currSum < 0) currSum = 0;
            // Add n to the current sum
            currSum += n;
            // Place the higher value between currSum and maxSum into maxSum
            maxSum = currSum > maxSum ? currSum : maxSum;
        }
        return maxSum;
    }
}